boa/src/syntax/ast/node/switch.rs

```diff
@@ -0,0 +1,85 @@
+//! Switch node.
+//!
+use super::{join_nodes, Node};
+use gc::{Finalize, Trace};
+use std::fmt;
+
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Clone, Debug, Trace, Finalize, PartialEq)]
+pub struct Switch {
+    val: Box<Node>,
+    cases: Box<[(Node, Box<[Node]>)]>,
```

Comment:
It's best if we implement a `Case` struct here, so that `cases` is just `Box<[Case]>`. This `Case` struct should have two members:
 - The case expression
 - The `StatementList`

Reply:
Done :)